pkg/network/ovn_kubernetes.go

```diff
@@ -212,6 +215,26 @@ func renderOVNKubernetes(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.Bo
 	return objs, nil
 }
 
+// returns the value of mode found in the openshift-ovn-kubernetes/gateway-mode-config configMap
+// if it exists, otherwise returns whatever the global OVN_GATEWAY_MODE is set to (shared)
+func GetGatewayMode(kubeClient client.Client) string {
+	cm := &corev1.ConfigMap{
+		TypeMeta:   metav1.TypeMeta{APIVersion: "v1", Kind: "ConfigMap"},
+		ObjectMeta: metav1.ObjectMeta{Name: "gateway-mode-config"},
+	}
+	nsn := types.NamespacedName{Namespace: "openshift-ovn-kubernetes", Name: "gateway-mode-config"}
```

Comment:
would this namespace exist at cluster create time? Plus your config map doesn't have this namespace in the hack example

Reply:
I don't know if the namespace will exist at this point or not. I'll have to test to check.

I fixed the hack/gateway-mode.yaml file to include the namespace.

Reply:
[This](https://github.com/openshift/cluster-network-operator/blob/master/docs/operands.md#general-notes) makes me think it'll be fine... specifically 

```
Most operand directories start with a 000-ns.yaml object to create the operand namespace before anything else is created.
```
Which makes me thing the ns should be there before `ovn-kubernetes.go` is ever touched